### DISCUSSION TITLE
Best way to emit async network request every time a log is handled?

### DISCUSSION THREAD

**Discussion by 0xForerunner** [tokio-rs/tracing#2788] [Q&A]
My logs need to be send to a server in real time. Basically every event needs to run an async fn which sends off the formatted logs. Is there a non hacky way to do this?

**Comment by davidbarsky** [comment]
I'm not sure what you might consider to be a "hacky way", but the general approach would be to send the formatted event over a channel where the receiver runs on a spawned task. That task would be responsible for running said async fn.

**Reply by hawkw** [reply]
This is the correct approach. Since a tracing event may be emitted from a synchronous _or_ asynchronous function, the `Subscriber::event` method may not be async. Instead, a channel should be used to bridge between the synchronous `Subscriber` callback and an asynchronous task for publishing log events. See [Tokio's documentation on bridging between synchronous and asynchronous code](https://tokio.rs/tokio/topics/bridging) for details.
